### PR TITLE
IBX-4004 Upgrade doc for 4.1 missing Fastly VCL changes needed for shielding support

### PR DESCRIPTION
**Pull request by MagdalenaZuba**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | [IBX-4004](https://issues.ibexa.co/browse/IBX-4004)
| Versions      | all

#### Checklist

- [ ] Text renders correctly
- [ ] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [ ] Code samples are working
- [ ] PHP code samples have been fixed with PHP CS fixer
- [ ] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

**Review by vidarl — CHANGES_REQUESTED** [review]
The changes added there are not the ones I had in mind.

The changes I am talking about are these:
https://doc.ibexa.co/en/latest/update_and_migration/from_2.5/to_3.3.latest/#vcl-configuration-for-fastly_1

So, anyone who upgrades from version lower than 3.3.24 and 4.1.6 needs to do those changes too

**Comment by MagdalenaZuba** [comment]
I will leave the changes to the snippet in this PR, as they cover the same configuration.

File: docs/update_and_migration/from_4.0/to_4.1.md [hunk 1]
```diff
@@ -166,6 +165,20 @@ Apply the following database update scripts:
     psql <database_name> < vendor/ibexa/installer/upgrade/db/postgresql/ibexa-4.1.0-to-4.1.1.sql
     ```
 
+### v4.1.6
+
+#### VCL configuration for Fastly
+
+Ibexa DXP now supports Fastly shielding. If you are using Fastly and want to use shielding, you need to update your VCL files.
```
Comment (vidarl) [inline docs/update_and_migration/from_4.0/to_4.1.md:172]: I think we should tell the users to upgrade nevertheless if they are planning to use shielding or not

```suggestion
The .vcl configuration files for Fastly has changed. Follow the upgrade description below to update then
```

File: docs/update_and_migration/from_4.0/to_4.1.md [hunk 2]
```diff
@@ -166,6 +165,20 @@ Apply the following database update scripts:
...
+
```
Comment (vidarl) [inline docs/update_and_migration/from_4.0/to_4.1.md:173]:
```suggestion
```

File: docs/update_and_migration/from_4.0/to_4.1.md [hunk 3]
```diff
@@ -166,6 +165,20 @@ Apply the following database update scripts:
...
+!!! tip
```
Comment (vidarl) [inline docs/update_and_migration/from_4.0/to_4.1.md:174]:
```suggestion
```

File: docs/update_and_migration/from_4.0/to_4.1.md [hunk 4]
```diff
@@ -166,6 +165,20 @@ Apply the following database update scripts:
...
+
+    Even if you do not plan to use Fastly shielding, it is recommended to update the VCL files for future compatibility.
```
Comment (vidarl) [inline docs/update_and_migration/from_4.0/to_4.1.md:176]:
```suggestion
```

File: docs/update_and_migration/from_4.0/to_4.1.md [hunk 5]
```diff
@@ -166,6 +165,20 @@ Apply the following database update scripts:
...
     ```
 
+### v4.1.6
+
+#### VCL configuration for Fastly
+
+Ibexa DXP now supports Fastly shielding. If you are using Fastly and want to use shielding, you need to update your VCL files.
+
+!!! tip
+
```
Comment (vidarl) [inline docs/update_and_migration/from_4.0/to_4.1.md:175]:
```suggestion
```

File: docs/update_and_migration/from_4.0/to_4.1.md [hunk 6]
```diff
@@ -166,6 +165,20 @@ Apply the following database update scripts:
...
+    Even if you do not plan to use Fastly shielding, it is recommended to update the VCL files for future compatibility.
+
+1. Locate the `vendor/ibexa/fastly/fastly/ez_main.vcl` file and update your VCL file with the recent changes.
+2. Do the same with `vendor/ibexa/fastly/fastly/ez_user_hash.vcl`.
+3. Upload a new `snippet_re_enable_shielding.vcl` snippet file, based on `vendor/ibexa/fastly/fastly/snippet_re_enable_shielding.vcl`.
+
```
Comment (vidarl) [inline docs/update_and_migration/from_4.0/to_4.1.md:181]:
```suggestion

Once the .vcl configuration has been updated, you may enable [Fastly Shielding](https://docs.fastly.com/en/guides/shielding) if you prefer.
```

File: docs/update_and_migration/from_4.1/update_from_4.1.md [hunk 7]
```diff
@@ -32,6 +32,20 @@ Run:
     composer require ibexa/commerce:[[= latest_tag_4_1 =]] --with-all-dependencies --no-scripts
     ```
 
+### VCL configuration for Fastly
+
+You can omit this step if you already performed it during update to v4.1.6.
+
+Ibexa DXP supports Fastly shielding from v4.1.6. If you are using Fastly and want to use shielding, you need to update your VCL files.
```
Comment (vidarl) [inline docs/update_and_migration/from_4.1/update_from_4.1.md:39]: I suggest the same changes as already mentioned ealier

**Review by vidarl — CHANGES_REQUESTED** [review]

File: docs/update_and_migration/from_4.0/to_4.1.md (hunk 1 above)
Comment (MagdalenaZuba) [reply docs/update_and_migration/from_4.0/to_4.1.md:172]:
```suggestion
The Fastly `.vcl` configuration files have changed. Follow the upgrade steps below to update them.
```

**Review by vidarl — APPROVED** [review]

**Review by DominikaK — APPROVED** [review]

**Pull request merged**